src/lib/inputs/InputNumber.svelte

```diff
@@ -0,0 +1,397 @@
+        border: none;
+        background: inherit;
+        color: inherit;
+        transition: all 0.1s ease-in-out;
+        &:focus {
+            outline: 3px solid currentColor;
+        }
+    }
+
+    .line input {
+        border-radius: 0;
+        border-bottom: 3px solid rgba(131, 131, 131, 0.5);
+        transition: border-color 0.3s ease-in-out;
+
+        &:focus {
+            outline: none;
+            border-bottom-color: currentColor;
+        }
+    }
+
+    .outline input {
+        border-radius: 50rem;
+        border: 3px solid rgba(131, 131, 131, 0.5);
+        transition: all 0.1s ease-in-out;
+        &:focus {
+            border: 3px solid currentColor;
+            background: rgba(255, 255, 255, 0.5);
+        }
+    }
+
+    .controls {
+        position: absolute;
+        top: 0;
+        right: 0;
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        align-items: center;
+        height: 100%;
+        width: 2.5rem;
+        .control {
+            display: flex;
+            justify-content: center;
+            align-items: center;
+            width: 100%;
+            height: 50%;
+            border: none;
+            background: transparent;
+            cursor: pointer;
+            outline: none;
+            svg {
+                width: 1rem;
+                height: 1rem;
+            }
+        }
+        .up {
+            &:hover {
+                transform: translateY(-1px);
+            }
+            &:active {
+                transform: translateY(1px);
+            }
+        }
+        .down {
+            &:hover {
+                transform: translateY(1px);
+            }
+            &:active {
+                transform: translateY(-1px);
+            }
+        }
+    }
+
+    // Sizes
+    .xsmall input {
+        padding: 0.125rem 0.25rem;
+    }
+
+    .small input {
+        padding: 0.25rem 0.5rem;
+    }
+
+    .medium input {
+        padding: 0.5rem 1rem;
+    }
+
+    .large input {
+        padding: 0.75rem 1.5rem;
+    }
+
+    .xlarge input {
+        padding: 1rem 2rem;
+    }
+
+    // Loading and error styles
+    .loading input,
+    .error input {
+        color: #fff;
+    }
+
+    .loading input {
+        outline: 3px solid var(--color);
+        outline-offset: -3px;
+        animation: double-pulse-gray 1.5s infinite ease-in-out;
+    }
+
+    .error input {
+        animation: double-pulse-red 1.5s infinite ease-in-out;
+    }
+
+    @keyframes double-pulse-gray {
+        0% {
+            box-shadow: 0 0 5rem 1rem rgba(0, 0, 0, 0) inset;
+        }
+        100% {
+            box-shadow: 0 0 0 0 rgba(0, 0, 0, 0.2) inset;
+        }
+    }
+
+    @keyframes double-pulse-red {
+        0% {
+            box-shadow: 0 0 5rem 1rem rgba(0, 0, 0, 0) inset;
+        }
+        100% {
+            box-shadow: 0 0 0 0 rgba(255, 0, 0, 0.5) inset;
+        }
+    }
+
+    .label {
+        font-size: 0.8rem;
```

Comment:
generally for font size accessibility rec would be 16px/1rem ... is this something a dev could change or would you consider making it an exposed scss variable?

Reply:
Hmmm, so you're suggestion that we add another prop for the label size?